### PR TITLE
Enable Social Sharing of Memes via URL

### LINKED ISSUE #311: Enable Rich Sharing of Content
Twitter & Facebook can show a preview when you post a link to their site if you use specific HTML `meta` tags in your returned HTML document. [Twitter example](https://dev.twitter.com/cards/markup)

This would be cool for memegen (and clients :wink:) because users would not have to download/upload images to share them on Social Media, but instead can just share links and they would show up on these sites.

### PR DESCRIPTION
**Pull request by jasich**
Closes #311

### CODE REVIEW

File: memegen/templates/image.html [hunk 1]
```diff
@@ -1,27 +1,33 @@
 <!DOCTYPE html>
 <html>
-<head>
+  <head>
+    <meta name="twitter:card" content="summary_large_image" />
+    <meta name="twitter:site" content="@jasich">
```
Comment (jacebrowning) [inline memegen/templates/image.html]: What do you think we should put here?

Comment (jacebrowning) [inline memegen/templates/image.html]: I just registered `@memegen_link` if needed.

Comment (jasich) [inline memegen/templates/image.html]: You should create a memegen twitter handle.  Depending on how engrained your branding is in the project you could make this a variable.

File: memegen/templates/image.html [hunk 2]
```diff
@@ -1,27 +1,33 @@
...
+    <meta name="twitter:title" content="memegen.link">
+    <meta name="twitter:description" content="{{ title }}">
+    <meta name="twitter:image" content="{{ src | safe }}">
+    <meta name="twitter:image:alt" content="oh no!">
```
Comment (jacebrowning) [inline memegen/templates/image.html]: Could the `alt` be the same as the title?

Comment (jasich) [inline memegen/templates/image.html]: Yeah I might remove it, I don't think it worked when the image failed to load.  Might only happen on specific error codes.

**Comment by jasich** [comment]
I'm wondering if for twitter if we make the image 280x150 it won't crop it, https://dev.twitter.com/cards/types/summary-large-image.

File: memegen/templates/image.html (hunk 1 above)
Comment (jacebrowning) [inline memegen/templates/image.html]: Let's change this to `@memegen_link`. Maybe that account can become a bot later that tweets popular memes. 😄

Comment (jasich) [inline memegen/templates/image.html]: Yeah that's what I was thinking

**Comment by jacebrowning** [comment]
> I'm wondering if for twitter if we make the image 280x150

Being able to request custom image sizes seems like a useful feature even outside of rendering for Twitter. I'm thinking that is outside the scope of this PR: https://github.com/jacebrowning/memegen/issues/313

**Comment by jasich** [comment]
@jacebrowning are you good with these changes now?

**Comment by jacebrowning** [comment]
Merge when ready!

**Pull request merged**